src/service/service.py

```diff
@@ -121,7 +122,10 @@ async def message_generator(user_input: StreamInput) -> AsyncGenerator[str, None
                 # Empty content in the context of OpenAI or Anthropic usually means
                 # that the model is asking for a tool to be invoked.
                 # So we only print non-empty content.
-                yield f"data: {json.dumps({'type': 'token', 'content': content})}\n\n"
+                if isinstance(content, str):
+                    yield f"data: {json.dumps({'type': 'token', 'content': content})}\n\n"
+                elif isinstance(content, AIMessageChunk):
+                    yield f"data: {json.dumps({'type': 'token', 'content': convert_message_content_to_string(content.content)})}\n\n"
```

Comment:
I think there's actually a bug here, because `content` is the List of dict, not an `AIMessageChunk`, so this always returns false.

I fixed it by getting rid of the if/else and replacing this section with one line:

```python
yield f"data: {json.dumps({'type': 'token', 'content': convert_message_content_to_string(content)})}\n\n"
```

It seems to work as expected after I make that change.

Reply:
You are correct, thanks for catching this!